src/harness/reference_models/ppa/ppa.py

```diff
@@ -0,0 +1,150 @@
+                                                         region=region_type).db_loss
+  index_cond, = np.where(install_param['eirpCapability'] + antenna_gain - db_loss > THRESHOLD_PER_10MHZ)
+  return index_cond.shape[0] / 5.
+
+
+def _HammingFilter(x, window_len=15):
+  s = np.r_[x[-(window_len / 2):], x, x[0:window_len / 2]]
+  w = np.hamming(window_len)
+  y = np.convolve(w / w.sum(), s, mode='valid')
+  return y
+
+
+def _GetPolygon(device):
+  install_param = device['installationParam']
+  if not 'eirpCapability' in install_param.keys():
+    install_param['eirpCapability'] = MAX_ALLOWABLE_EIRP_PER_10_MHZ_CAT_A \
+      if device['cbsdCategory'] == 'A' else MAX_ALLOWABLE_EIRP_PER_10_MHZ_CAT_B
+
+  # Get all the Function Partials
+  # Compute all the Points in 0-359 every 200m upto 40km
+  distance = np.arange(0.2, 40.2, 0.2)
+  azimuths = np.arange(0.0, 360.0)
+  latitudes, longitudes, _ = zip(*[vincenty.GeodesicPoints(install_param['latitude'],
+                                                           install_param['longitude'],
+                                                           distance, azimuth)
+                                   for azimuth in azimuths])
+  # Compute the Gain for all Direction
+  antenna_gain = antenna.GetStandardAntennaGains(azimuths,
+                                                 install_param['antennaAzimuth']
+                                                 if 'antennaAzimuth' in install_param.keys() else None,
+                                                 install_param['antennaBeamwidth'],
+                                                 install_param['antennaGain'])
+  # Compute the Path Loss, and contour based on Gain and Path Loss Comparing with Threshold
+  # Smoothing Contour using Hamming Filter
+  contour_pts = _HammingFilter([_CalculateDbLossForEachPointAndGetContour(install_param, gain, lat, lon)
+                                for lat, lon, gain in zip(latitudes, longitudes, antenna_gain)])
+  # Generating lat, lon for Contour
+  contour_lats, contour_lons, _ = zip(*[vincenty.GeodesicPoint(install_param['latitude'],
+                                                               install_param['longitude'], cn, az)
+                                        for cn, az in zip(contour_pts, azimuths)])
+  return geometry.Polygon(zip(contour_lons, contour_lats)).buffer(0)
+
+
+def _ClipPpaByCensusTract(contour_union, pal_records):
+  # Get the Census Tract for Each Pal Record and Convert it to Shapely Geometry
+  census_tracts_for_pal = [geometry.shape(census_tract_driver.GetCensusTract(pal['license']
+                                                                             ['licenseAreaIdentifier'])
+                                          ['features'][0]['geometry']).buffer(0) for pal in pal_records]
+  census_tracts_union = ops.cascaded_union(census_tracts_for_pal)
+  return contour_union.intersection(census_tracts_union)
+
+
+def _ConvertToGeoJson(ppa_polygon):
```

Comment:
This and previous function could have a onliner docstring description.
I know they are intended as private function, but a one liner is still useful, for example:
"""Converts a shapely Polygon or MultiPolygon into a GeoJSON feature collection"""

and for previous
""" Clip a PPA 'contour_union' zone (shapely.MultiPolygon) with the census tracts defined by a sequence of 'pal_records'."""